client/interledger/js/components/asset_row.js

```diff
@@ -9,87 +12,177 @@ import AssetDetail from '../../../lib/js/react/components/asset_detail';
+        if (this.getOperation() !== 'transfer') {
+            this.intervalId = window.setInterval(() => {
+                this.setExpiryTime();
+            }, 1000);
+        }
+    },
+
+    componentWillUnmount() {
+        window.clearInterval(this.intervalId);
+    },
+
+    setExpiryTime() {
+        const {
+            asset
+        } = this.props;
+        const expires = moment(parseInt(parseFloat(asset.expiryTime) * 1000, 10));
+        const expiresIn = moment.utc(expires.diff(moment.utc()));
+        this.setState({
+            expiresIn: expiresIn > 0 ? expiresIn : -1
+        });
+    },
+
     handleAssetClick() {
         const {
-            activeAccount,
+            account,
             handleAccountClick,
             asset,
             handleAssetClick
         } = this.props;
 
         safeInvoke(handleAssetClick, asset);
-        safeInvoke(handleAccountClick, activeAccount);
+        safeInvoke(handleAccountClick, account);
     },
 
-    handleEscrowClick(selectedAccount) {
+    handleActionClick(selectedAccount) {
         const {
-            asset,
-            activeAccount
+            account,
+            asset
         } = this.props;
-        
+
         const idToTransfer = {
             txid: asset.id,
             cid: 0
         };
 
-        const payloadToPost = {
-            'source': activeAccount,
-            'to': selectedAccount
-        };
+        if (asset.type === 'single-owner') {
+            AssetActions.escrowAsset({
+                idToTransfer,
+                payloadToPost: {
+                    source: account,
+                    to: selectedAccount
+                }
+            });
+        } else if (asset.type === 'multi-owner') {
+            AssetActions.fulfillEscrowAsset({
+                idToTransfer,
+                payloadToPost: {
+                    source: account,
+                    to: account
+                }
+            });
+        }
+    },
 
-        AssetActions.escrowAsset({
-            idToTransfer,
-            payloadToPost
-        });
+    getOperation() {
+        const {
+            account,
+            asset
+        } = this.props;
 
-        this.setState({ inEscrow: true });
+        let operation = 'transfer';
+        if (asset.hasOwnProperty('executeCondition') &&
+            account.vk === asset.executeCondition.public_key) {
+            operation = 'execute';
+        } else if (asset.hasOwnProperty('abortCondition') &&
+            account.vk === asset.abortCondition.public_key) {
+            operation = 'abort';
+        }
+        return operation;
     },
 
     render() {
         const {
+            account,
             accountList,
-            activeAccount,
+            actionMap,
             asset,
             isActive
         } = this.props;
 
         const {
-            inEscrow
+            expiresIn
         } = this.state;
 
+        const inBacklog = asset.hasOwnProperty('assignee');
```

Comment:
I remember we added some comments about this somewhere else earlier... maybe it'd be good to put this in a utils method with those comments?